### ISSUE TITLE
Sidebar headers are not visible enough

### ISSUE THREAD

**Issue by nickvergessen** [nextcloud-libraries/nextcloud-vue#4785]
Have a "dark" theming color and use dark mode

## Expected

The sidebar is clearly visible

## Actual
![grafik](https://github.com/nextcloud-libraries/nextcloud-vue/assets/213943/e70039a3-f689-432c-a599-6891461ceea6)

**Comment by nickvergessen** [comment]
Comments in the chat:

- @nimishavijay 
  > You are right, that color combination does not have enough contrast, the fallback colors should be triggered for such a dark color in dark mode. cc @marcoambrosini @szaimen 

- @susnux 
  > I think we probably need to go the same way like with the button component which enforces primary style for checked state
